test/src.test.ts

```diff
@@ -0,0 +1,244 @@
+					fee : '0'
+				},
+				{
+					id : '0xce07CbB7e054514D590a0262C93070D838bFBA2e',
+					url : 'https://scratchy.adex.network',
+					fee : '0'
+				}
+			],
+			minTargetingScore: 0
+		}
+	}
+}
+
+test('Normalize URL tests', (t) => {
+	const ipfsUrl = 'ipfs://QmcUVX7fvoLMM93uN2bD3wGTH8MXSxeL8hojYfL2Lhp7mR'
+	const normalUrl = 'https://www.xxxtentacion.rip/'
+
+	t.equals(normalizeUrl(ipfsUrl), IPFS_GATEWAY + 'QmcUVX7fvoLMM93uN2bD3wGTH8MXSxeL8hojYfL2Lhp7mR', 'Slices the ipfs:// part successfully')
+	t.equals(normalizeUrl(normalUrl), normalUrl, 'Doesn\'t change normal URL')
+	t.end()
+})
+
+test('Calculate Target Score tests', (t) => {
+	const targetTags1 = [{tag: 'test', score: 10}]
+	const targetTags2 = [{tag: 'test', score: 30}]
+	const targetTags3 = [{tag: 'test2', score: 30}]
+	const targetTags4 = [{tag: 'test', score: 20}, { tag: 'test2', score: 100 }]
+	const targetTags5 = [{tag: 'test', score: 50}, { tag: 'test3', score: 50 }]
+
+	t.equals(calculateTargetScore(targetTags1, targetTags2), 300, 'Calculates correct target score for 2 matches')
+	t.equals(calculateTargetScore(targetTags1, targetTags3), 0, 'Calculates correct target score for no matches')
+	t.equals(calculateTargetScore(targetTags3, targetTags5), 0, 'Calculate correct target score for no matches on tag array with more than one element')
+	t.equals(calculateTargetScore(targetTags1, targetTags4), 200, 'Calculates correct target score with one match')
+	t.equals(calculateTargetScore(targetTags4, targetTags5), 1000, 'Calculates correct target score with 1 match in arrays with more than one element')
+	t.equals(calculateTargetScore(targetTags4, targetTags4), 10400, 'Calculates correct target score for more than one tag match')
+	t.equals(calculateTargetScore([], []), 0, 'Shows 0 if no tags at all')
+	t.end()
+})
+
+test('Get HTML tests', (t) => {
+	const otherInfo = {
+		unit: {
+			ipfs: 'ipfs://QmcUVX7fvoLMM93uN2bD3wGTH8MXSxeL8hojYfL2Lhp7mR',
+			targetUrl: 'https://xxxtentacion.com',
+			mediaUrl: 'ipfs://QmcUVX7fvoLMM93uN2bD3wGTH8MXSxeL8hojYfL2Lhp7mR'
+		},
+		channelId: '0x0',
+		validators: [{ url: 'https://tom.adex.network' }, { url: 'https://jerry.adex.network' }]
+	}
+
+	const resultHTML = `<a href="${otherInfo.unit.targetUrl}" target="_blank" rel="noopener noreferrer"><img src="https://ipfs.adex.network/ipfs/QmcUVX7fvoLMM93uN2bD3wGTH8MXSxeL8hojYfL2Lhp7mR" data-event-body='{"events":[{"type":"IMPRESSION","publisher":"0x13e72959d8055dafa6525050a8cc7c479d4c09a3","adUnit":"ipfs://QmcUVX7fvoLMM93uN2bD3wGTH8MXSxeL8hojYfL2Lhp7mR"}]}' alt="AdEx ad" rel="nofollow" onload="fetch('${otherInfo.validators[0].url}/channel/${otherInfo.channelId}/events', { method: 'POST', headers: { 'content-type': 'application/json' }, body: this.dataset.eventBody });fetch('${otherInfo.validators[1].url}/channel/${otherInfo.channelId}/events', { method: 'POST', headers: { 'content-type': 'application/json' }, body: this.dataset.eventBody })" width="300" height="100" ></a>`
```

Comment:
* Maybe it can be done by parsing and validating the HTML, and then validate each attribute.
* need to do this for image and video media type